jitsi-utils-kotlin/src/main/kotlin/org/jitsi/utils/queue/QueueStatistics.kt

```diff
@@ -0,0 +1,227 @@
+/*
+* Copyright @ 2018 - present 8x8, Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package org.jitsi.utils.queue
+
+import org.jitsi.utils.OrderedJsonObject
+import org.jitsi.utils.stats.BucketStats
+import org.jitsi.utils.stats.RateStatistics
+import org.json.simple.JSONObject
+import java.util.Collections
+import java.util.concurrent.atomic.LongAdder
+import java.util.IdentityHashMap
+import java.util.concurrent.ConcurrentHashMap
+
+class QueueStatisticsInternal(queueSize: Int) {
+    /**
+     * Total packets added to the queue.
+     */
+    private val totalPacketsAdded = LongAdder()
+
+    /**
+     * Total packets removed to the queue.
+     */
+    private val totalPacketsRemoved = LongAdder()
+
+    /**
+     * Total packets dropped from the queue.
+     */
+    private val totalPacketsDropped = LongAdder()
+
+    /**
+     * The time the first packet was added.
+     */
+    private var firstPacketAddedMs: Long = -1
+
+    /**
+     * Statistics about queue lengths
+     */
+    private val queueLengthStats = BucketStats(lengthBucketSizes(queueSize), "_queue_size_at_remove", "")
+
+    /**
+     * Statistics about the time that packets were waiting in the queue.
+     */
+    private val queueWaitStats = if (QueueStatistics.trackTimes) BucketStats(waitBucketSizes, "_queue_wait_time_ms", " ms") else null
+
+    /**
+     * Gets a snapshot of the stats in JSON format.
+     */
+    val stats: OrderedJsonObject
+        get() {
+            val stats = OrderedJsonObject()
+            val now = System.currentTimeMillis()
+            stats["added"] = totalPacketsAdded.sum()
+            stats["removed"] = totalPacketsRemoved.sum()
+            stats["dropped"] = totalPacketsDropped.sum()
+            val duration = (now - firstPacketAddedMs) / 1000.0
```

Comment:
nit: name `durationSecs` or, even better, use `Instant`s and a real `Duration`! :)